### PR TITLE
Use redux for username in masthead toolbar

### PR DESCRIPTION
**Pull request by TheRealJon**
Refactor masthead toolbar to use redux for the username, when available. Also consolidated mobile conditionals into a single block in renderMenu function.

### CODE REVIEW

File: frontend/public/components/masthead-toolbar.jsx [hunk 1]
```diff
@@ -244,4 +227,7 @@ class MastheadToolbar_ extends React.Component {
   }
 }
 
-export const MastheadToolbar = connectToFlags(FLAGS.AUTH_ENABLED, FLAGS.OPENSHIFT, FLAGS.CLUSTER_UPDATES_AVAILABLE)(MastheadToolbar_);
+export const MastheadToolbar = _.flow(
+  connect(userStateToProps),
+  connectToFlags(FLAGS.AUTH_ENABLED, FLAGS.OPENSHIFT, FLAGS.CLUSTER_UPDATES_AVAILABLE),
+)(MastheadToolbar_);
```
Comment (spadgett) [inline frontend/public/components/masthead-toolbar.jsx]: I'd avoid connecting twice if possible. Instead you could use

```js
import { FLAGS, connectToFlags, flagPending, stateToProps as flagStateToProps } from '../features';

const mastheadToolbarStateToProps = state => {
  const desiredFlags = [FLAGS.AUTH_ENABLED, FLAGS.OPENSHIFT, FLAGS.CLUSTER_UPDATES_AVAILABLE];
  const flags = flagStateToProps(desiredFlags, state);
  const user = state.UI.get('user');
  return { flags, user };
};
```

File: frontend/public/components/masthead-toolbar.jsx [hunk 2]
```diff
@@ -39,25 +39,18 @@ class MastheadToolbar_ extends React.Component {
...
-          this.setState({ username: _.get(user, 'fullName') || user.metadata.name });
-        })
-        .catch(() => {
-          this.setState({ username: null });
-        });
-    } else {
-      this.setState({ username: authSvc.userID() ? authSvc.name() : null });
+    const { flags, user } = this.props;
+    if (!flags[FLAGS.OPENSHIFT]) {
+      this.setState({ username: authSvc.name() });
```
Comment (spadgett) [inline frontend/public/components/masthead-toolbar.jsx]: We probably want to skip this if not `AUTH_ENABLED`

Comment (TheRealJon) [reply frontend/public/components/masthead-toolbar.jsx]: So no username should be shown at all if auth is disabled?

Comment (spadgett) [reply frontend/public/components/masthead-toolbar.jsx]: Sorry, I'm wrong. Only the logout action is disabled if auth is disabled.

**Pull request merged**